cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -275,56 +280,127 @@ void accumulate_vertex_results(
     thrust::make_zip_iterator(distances.begin(), sigmas.begin(), deltas.begin()),
     dst_properties.mutable_view());
 
-  // FIXME: To do this efficiently, I need a version of
-  //   per_v_transform_reduce_outgoing_e that takes a vertex list
-  //   so that we can iterate over the frontier stack.
-  //
-  //   For now this will do a O(E) pass over all edges over the diameter
-  //   of the graph.
-  //
+  // Pre-allocate reusable buffers to avoid repeated allocations
+  auto max_frontier_size = transform_reduce_v(
+    handle,
+    graph_view,
+    distances.begin(),
+    [] __device__(auto, auto d) { return (d == invalid_distance) ? vertex_t{0} : vertex_t{1}; },
+    vertex_t{0},
+    reduce_op::plus<vertex_t>{},
+    do_expensive_check);
+  
+  rmm::device_uvector<vertex_t> reusable_vertex_buffer(max_frontier_size, handle.get_stream());
+  rmm::device_uvector<weight_t> reusable_delta_buffer(max_frontier_size, handle.get_stream());
+  rmm::device_uvector<weight_t> reusable_centrality_buffer(max_frontier_size, handle.get_stream());
+
   // Based on Brandes algorithm, we want to follow back pointers in non-increasing
   // distance from S to compute delta
   //
   for (vertex_t d = diameter; d > 1; --d) {
-    per_v_transform_reduce_outgoing_e(
+    // Clear deltas array for this iteration
+    detail::scalar_fill(handle, deltas.data(), deltas.size(), weight_t{0});
```

Comment:
deltas.size() == graph_view.local_vertex_partition_range_size(), right?

You don't need to reset deltas in every iteration. This wastes a lot of computing. You need to find the deltas only for the vertices in the frontier.